### PR TITLE
BCDA-3607 Parallelized Smoke Test and Fix Bulk Lite

### PR DESCRIPTION
**Pull request by CosmicSyntax**
### Fixes [BCDA-3607](https://jira.cms.gov/browse/BCDA-3607)
As an effort to speed up the testing and release process, it has been suggested to parallelize the smoke test. This ticket aims to parallelize a number of calls made by bcda_client.go. This ticket also explored parallelizing the integration smoke-test in jenkins.
### Proposed Changes
Bash / sh has the feature to send a task into a separate child process through "&". We can add this to the end of commands we want to parallelize. We need to ensure however that we collect the PID of that child process and wait for it specifically (e.g. wait \<PID\> not just wait). By doing this, we make sure if any of the child processes fail, the parent process will exit with a code other than 0.
### Change Details
We implemented the proposed changes mentioned above to the following files:
1. bulk_data_requests.sh 
2. bulk_data_requests_lite.sh 
To keep some order in the logs and not to over-burden local test, we removed parallelization in the smoke_test.sh.
A success run of the smoke-test does indeed return an exit code of 0, otherwise an exit code of 2.
Screen Shot 2021-07-02 at 12 51 35 AM
Screen Shot 2021-07-02 at 1 31 24 AM
To simulate a true failure, we substituted one of the calls in bulk_data_requests.sh with a bogus secret.
Screen Shot 2021-07-02 at 1 26 07 AM
We have seen significant improvements in the speed of the smoke-test locally, but gains on GitHub Actions (GA) are minimal. We suspect that the VMs used for the job in GA are low in CPU cores, where parallelize jobs end up running mostly sequentially.

A proven method of improving the speed of GA is to launch our tests in separate jobs, as seen here:
https://github.com/CMSgov/bcda-app/actions/runs/991957013
One down side of this is that each job builds its own stack. We tried to leverage GA cache, but preliminary results showed that while it is faster than the current method, it's slower than just launching separate jobs, as seen here:
https://github.com/CMSgov/bcda-app/actions/runs/992333508

Finally, we discovered that Jenkins integration smoke-test is already leveraging parallelism.
https://github.com/CMSgov/bcda-ops/blob/98292470634c612ff815d26dbcd9d3f1f58b30ba/jenkins_files/Jenkinsfile.smoke_test#L504
### Security Implications
- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change
### Acceptance Validation
The attempt to speed up smoke-test does indeed speed up locally, and there is no holes in the implementation.
### Feedback Requested
Any feedback is good feedback.

### CODE REVIEW

**Review by msnook — CHANGES_REQUESTED** [review]
This is cool.  When looking at [Github Actions](https://github.com/CMSgov/bcda-app/runs/2969305768?check_suite_focus=true) to examine the performance, I noticed that at [line 473](https://github.com/CMSgov/bcda-app/runs/2969305768?check_suite_focus=true#step:6:473) we get the error below.  Was this happening before?  It looks like some errors are happening (or maybe some tests are not running) and they aren't being triggered as a failure... is this Postman?

```
[42999] Failed to execute script docker-compose
Traceback (most recent call last):
  File "dockerpty/pty.py", line 334, in start
  File "dockerpty/pty.py", line 367, in _hijack_tty
  File "dockerpty/io.py", line 59, in select
  File "dockerpty/io.py", line 351, in fileno
  File "dockerpty/io.py", line 103, in fileno
  File "socket.py", line 638, in fileno
ValueError: I/O operation on closed file.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 81, in main
  File "compose/cli/main.py", line 203, in perform_command
  File "compose/metrics/decorator.py", line 18, in wrapper
  File "compose/cli/main.py", line 611, in exec_command
  File "dockerpty/pty.py", line 338, in start
  File "dockerpty/io.py", line 32, in set_blocking
  File "dockerpty/io.py", line 351, in fileno
  File "dockerpty/io.py", line 103, in fileno
  File "socket.py", line 638, in fileno
ValueError: I/O operation on closed file.
[43029] Failed to execute script docker-compose
Traceback (most recent call last):
  File "dockerpty/pty.py", line 334, in start
  File "dockerpty/pty.py", line 367, in _hijack_tty
  File "dockerpty/io.py", line 59, in select
  File "dockerpty/io.py", line 351, in fileno
  File "dockerpty/io.py", line 103, in fileno
  File "socket.py", line 638, in fileno
ValueError: I/O operation on closed file.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 81, in main
  File "compose/cli/main.py", line 203, in perform_command
  File "compose/metrics/decorator.py", line 18, in wrapper
  File "compose/cli/main.py", line 611, in exec_command
  File "dockerpty/pty.py", line 338, in start
  File "dockerpty/io.py", line 32, in set_blocking
  File "dockerpty/io.py", line 351, in fileno
  File "dockerpty/io.py", line 103, in fileno
  File "socket.py", line 638, in fileno
ValueError: I/O operation on closed file.
```

**Comment by CosmicSyntax** [comment]
@msnook Looks like it's been happening before, as seen here:
https://github.com/CMSgov/bcda-app/runs/2964172603?check_suite_focus=true
and here:
https://github.com/CMSgov/bcda-app/runs/2825023932?check_suite_focus=true

Going to look into why this may be happening.

**Comment by CosmicSyntax** [comment]
Postman issue has been fixed.
As seen here:
https://github.com/actions/virtual-environments/issues/808
TLDR: It may be an issue with how Github Action VM is configured. By adding the "T" flag option, the issue goes away. The "T" flag disables pseudo-tty allocation..

**Review by msnook — APPROVED** [review]
💯
